pulsar/internal/connection.go

```diff
@@ -327,8 +327,8 @@ func (c *connection) doHandshake() bool {
 }
 
 func (c *connection) waitUntilReady() error {
-	c.Lock()
-	defer c.Unlock()
+	c.cond.L.Lock()
```

Comment:
Curious what's the difference between doing `c.Lock()` vs `c.cond.L.Lock()`? Same for the changeState function? From looking at the code they should share the same lock?

Reply:
They're two different mutexes. See https://github.com/apache/pulsar-client-go/pull/663#issuecomment-963931326 for an explanation.

Reply:
I agree that `waitUntilReady()` and `changeState()` should share the same lock. I think the correct lock to use is `c.cond.L`.

Reply:
Nope, I was wrong! See https://github.com/apache/pulsar-client-go/pull/663#issuecomment-963966453 which proves that `c.Lock()` and `c.cond.L.Lock()` are the same thing.